### PR TITLE
[WIP] Core Alloy Migration

### LINKED ISSUE #5216: Meta: Alloy migration
Tracking issue for the Alloy Migration.

### Migration progress per-crate:
- [x] `debugger` https://github.com/foundry-rs/foundry/pull/5854
- [x] `fmt` https://github.com/foundry-rs/foundry/pull/5851
- [x] `cli`
- [x] `common`
- [x] `config` partially migrated
- [x] `cast`
  - [x] primitive types migrated
  - [x] signer/provider types migrated
- [x] `macros`
- [x] `abi`
- [x] `evm`
  - [x] Cheatcodes migrated
  - [x] ABI-related types migrated
  - [x] primitive types migrated https://github.com/foundry-rs/foundry/pull/5768
- [x] `anvil-core`
  - [x] https://github.com/foundry-rs/foundry/issues/6715
- [x] `anvil`
  - [x] txpool/executor logic migrated to alloy
  - [x] RPC types migrated
  - [x] primitive types migrated (partially migrated so far) https://github.com/foundry-rs/foundry/pull/5768
  - [x] RLP related types migrated
  - [x] Tests migrated to alloy
- [x] `utils` 
  - [x] primitive types migrated
- [x] `forge`
  - [x]  primitive types migrated https://github.com/foundry-rs/foundry/pull/5873
  - [x] forge-script migrated
- [x] `test-utils migrated https://github.com/foundry-rs/foundry/pull/5874
- [x] `chisel`

### provider/rpc types migration progress per-crate (only ones with usage of these are mentioned):
- [x] `debugger`
- [x] `fmt`
- [x] `cli`
- [x] `common`
- [x] `config`
- [x] `cast`
- [x] `abi`
- [x] `evm`
- [x] `anvil`
- [x] `forge`

done with https://github.com/foundry-rs/foundry/pull/6219

### Reth-related milestones
- [x] Yank reth revm inspectors into their own repo to remove a dependency on reth: https://github.com/paradigmxyz/reth/issues/5956

### soft/hard Blockers
- [x] `Log` alloy-primitives implementation, needed for a complete `anvil` & `evm` migration. https://github.com/alloy-rs/core/pull/271 (Soft blocker)
- [x] `ethers-solc` needs to move to its own standalone repo. (Soft blocker for now)
- [x] `ethers-etherscan` and other verification/etherscan related utilities need to either move to their own repo or integrate into an existing one. (Soft blocker)

### Followups
- #6104 
- #6102 
- #6098 
- https://github.com/foundry-rs/block-explorers/issues/8

### PR DESCRIPTION
**Pull request by Evalir**
Base branch for the Alloy Migration. Partially resolves #5216.

** This branch must be kept up to date with master and rebased at all times. **

All alloy migration PRs must be stacked against this branch, and will be merged in order (if required). 

**SCOPING IN PROGRESS**
- [ ] Migrate `evm` crate:
  - [ ] Executors

### CODE REVIEW

**Comment by Evalir** [comment]
need to bump alloy in revm to a version with https://github.com/alloy-rs/core/pull/244 cc @DaniPopes

File: crates/evm/src/executor/fork/database.rs [hunk 1]
```diff
@@ -285,7 +286,9 @@ mod tests {
         let backend = SharedBackend::spawn_backend(Arc::new(provider), db.clone(), None).await;
 
         let mut db = ForkedDatabase::new(backend, db);
-        let address = B160::random();
+        let address = Address::from_word(b256!(
+            "000000000000000000000000d8da6bf26964af9d7eed9e03e53415d37aa96045"
+        ));
```
Comment (DaniPopes) [inline crates/evm/src/executor/fork/database.rs:291]: FYI we have [`B*::random`](https://docs.rs/alloy-primitives/latest/alloy_primitives/struct.Address.html#method.random) but it's behind the `getrandom` feature flag, and also there's an [`address!`](https://docs.rs/alloy-primitives/latest/alloy_primitives/macro.address.html) macro

Comment (Evalir) [reply crates/evm/src/executor/fork/database.rs:291]: sweet nice—will change to use that! rn i'm in the process of just making the PR compile without the migration itself (which is going to be stacked, per crate ideally)

**Comment by Evalir** [comment]
Closing in favor of using #5768 as base

**Pull request closed without merging**